comp/core/status/statusimpl/status_api_endpoints_test.go

```diff
@@ -0,0 +1,239 @@
+			},
+		},
+		{
+			testDesc:    "json format",
+			method:      "GET",
+			routerPath:  "/status",
+			testedPath:  "/status?format=json",
+			httpHandler: provider.APIGetStatus.Provider.HandlerFunc(),
+			expectedBody: func() []byte {
+				status, err := provider.Comp.GetStatus("json", false)
+				require.NoError(t, err)
+				return status
+			}(),
+			expectedCode: http.StatusOK,
+			additionalTests: func(t *testing.T, rr *httptest.ResponseRecorder) {
+				require.Equal(t, "application/json", rr.Header().Get("Content-Type"))
+			},
+		},
+		{
+			testDesc:    "unknown format",
+			method:      "GET",
+			routerPath:  "/status",
+			testedPath:  "/status?format=unknown",
+			httpHandler: provider.APIGetStatus.Provider.HandlerFunc(),
+			expectedBody: func() []byte {
+				status, err := provider.Comp.GetStatus("text", false)
+				require.NoError(t, err)
+				return status
+			}(),
+			expectedCode: http.StatusOK,
+			additionalTests: func(t *testing.T, rr *httptest.ResponseRecorder) {
+				require.Equal(t, "text/plain", rr.Header().Get("Content-Type"))
+			},
+		},
+		{
+			testDesc:    "with error",
+			method:      "GET",
+			routerPath:  "/status",
+			testedPath:  "/status?format=text",
+			httpHandler: providerWithError.APIGetStatus.Provider.HandlerFunc(),
+			expectedBody: func() []byte {
+				status, err := providerWithError.Comp.GetStatus("text", false)
+				require.NoError(t, err)
+				return status
+			}(),
+			expectedCode: http.StatusOK,
+			additionalTests: func(t *testing.T, rr *httptest.ResponseRecorder) {
+			},
+		},
+		{
+			method:      "GET",
+			routerPath:  "/sections",
+			testedPath:  "/sections",
+			httpHandler: provider.APIGetSectionList.Provider.HandlerFunc(),
+			expectedBody: func() []byte {
+				res, err := json.Marshal(provider.Comp.GetSections())
+				require.NoError(t, err)
+				return res
+			}(),
+			expectedCode: http.StatusOK,
+			additionalTests: func(t *testing.T, rr *httptest.ResponseRecorder) {
+				require.Equal(t, "application/json", rr.Header().Get("Content-Type"))
+			},
+		},
+		{
+			testDesc:    "with header section",
+			method:      "GET",
+			routerPath:  "/{component}/status",
+			testedPath:  "/header/status",
+			httpHandler: provider.APIGetSection.Provider.HandlerFunc(),
+			expectedBody: func() []byte {
+				status, err := provider.Comp.GetStatusBySections([]string{"header"}, "text", false)
+				require.NoError(t, err)
+				return status
+			}(),
+			expectedCode: http.StatusOK,
+		},
+		{
+			testDesc:    "with unknown section text format",
+			method:      "GET",
+			routerPath:  "/{component}/status",
+			testedPath:  "/unknown/status",
+			httpHandler: provider.APIGetSection.Provider.HandlerFunc(),
+			expectedBody: func() []byte {
+				_, err := provider.Comp.GetStatusBySections([]string{"unknown"}, "text", false)
+				require.Error(t, err)
+
+				return []byte(fmt.Sprintf("Error getting status. Error: %v.\n", err))
+			}(),
+			expectedCode: http.StatusInternalServerError,
```

Comment:
Not for this PR, but I think `StatusInternalServerError` is not the correct status code for this request. 
Since this scenario is a invalid request from the customer that requested, I believe in that case we should use the 400 error code. 

Since that would require a bit of refactoring in the status component, let's do that work on a next PR 😄

Reply:
I agree with you ! I wrote it in a card for later.